### PR TITLE
Fixing .NET test failures and adding Gherkin/.NET support for VertexProperty.

### PR DESCRIPTION
**Pull request by mikepersonick**
https://issues.apache.org/jira/browse/TINKERPOP-2688
  https://issues.apache.org/jira/browse/TINKERPOP-2689

### CODE REVIEW

**Comment by FlorianHockmann** [comment]
Thanks for this great contribution to Gremlin.NET, @mikepersonick! I just added two nitpick comments about C# style, but this is good to go from my side with or without addressing them.

VOTE +1

**Comment by spmallette** [comment]
Adding `VertexProperty` to the test suite is most excellent - thanks.  

VOTE +1

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Process/Remote/RemoteStrategyTests.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿﻿#region License
```
[the remark this answers is not in the document]
Comment (mikepersonick) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Process/Remote/RemoteStrategyTests.cs:1]: Is this a known thing? Rider was not able to compile those source files for me bc of it. Is there an IDE setting that can fix it?

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs [hunk 2]
```diff
@@ -133,6 +137,70 @@ public ScenarioData(IMessageSerializer messageSerializer)
...
+                 * It will also take care of wrapping propVal in the appropriate Numeric format. We must do this in
+                 * case a Vertex has multiple properties with the same key and number value but in different numeric
+                 * type spaces (rare, but possible, and presumably something we may want to write tests around).
+                 */
+                string groovy = @"
+                    { vp ->
+                          def result = vp.element().value('name') + '-' + vp.key() + '->'
+                          def value = vp.value()
+                          def type = ''
+                          switch(value) {
```
[the remark this answers is not in the document]
Comment (mikepersonick) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Gherkin/ScenarioData.cs:157]: Some poor soul has to do the same in Javascript and Python...

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Process/Remote/RemoteStrategyTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (FlorianHockmann) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Process/Remote/RemoteStrategyTests.cs:1]: I at least remember that we had some problems with this already in the past. I think some IDE (maybe Visual Studio) added the BOM byte to files which resulted in problems for others. I think we should remove it everywhere.
We could add an `.editorconfig` where we specify the charset as `utf-8` for all `*.cs` files. Most editors / IDEs support `.editorconfig` (a plugin might be necessary though) which should ensure that the BOM byte doesn't get added back. We can also use a GitHub action to enforce that the `.editorconfig` is honored.
This issue contains more details: editorconfig/editorconfig#297.

**Pull request merged**